pkg/controller/backup_controller.go

```diff
@@ -161,6 +180,10 @@ func NewBackupController(
 		},
 	)
 
+	if features.IsEnabled(velerov1api.UploadProgressFeatureFlag) {
+		// Start the background task that checks upload progress and finalizes backups
+		go c.checkAndFinalizeBackupsLoop()
```

Comment:
The status of the Backup is updated in this goroutine but also in the main syncHandler for this controller. It's confusing to follow the flow and understand where the status is updated. Is there any way that the upload status can be checked in the goroutine, but make the main reconciliation loop responsible for updating the status of the Backup?

Reply:
I don't really see how.  The main reconciliation loop only kicks in on changes in the backup resource so it won't get triggered by the changes in the number of snapshots uploaded.